provider/consulcatalog/consul_catalog.go

```diff
@@ -156,14 +155,8 @@ func (p *Provider) watch(configurationChan chan<- types.ConfigMessage, stop chan
 	defer close(stopCh)
 	defer close(watchCh)
 
-	for {
-		select {
-		case <-stop:
-			return nil
-		case index, ok := <-watchCh:
-			if !ok {
-				return errors.New("consul service list nil")
```

Comment:
In fact, as we return an error here, the caller (https://github.com/ortz/traefik/blob/257d278191f547db00c5cd1fccd008324e5b8a60/provider/consulcatalog/consul_catalog.go#L109) will automatically retry because we use backoff.RetryNotify (https://github.com/ortz/traefik/blob/257d278191f547db00c5cd1fccd008324e5b8a60/provider/consulcatalog/consul_catalog.go#L111)